### ISSUE TITLE
How to use Form tag and submit to a post-route?

### ISSUE THREAD

**Issue by sjorsvanheuveln** [reactstrap/reactstrap#531]
Hi there, I'm new to reactstrap and I was wondering how to setup a form that posts to a post-route on submit. I couldn't find an example in the documentation that did this. So I'm looking for an equivalent of this:

```
<form action='/signup/insert' method = 'POST'>
    <div class="form-group">
        <input id ='signup' type='text' name='uid' placeholder='Username'><br>
        <input id ='signup' type='password' name='pwd' placeholder='Password'><br>
        <input id ='signup' type='text' name='email' placeholder='E-mail address'><br>
        <button id = 'switch' type='submit'>Sign Up</button>
     </div>
 </form>
```

**Comment by TheSharpieOne** [comment]
Just import `Form` from reactstrap and change `form` to `Form`. Any attributes you provided `Form` will be passed to the `<form>` it renders. ([Almost] Every reactstrap component will pass extra props/attributes down to the rendered HTML.)

Here is your example, but with reactstrap components (plus, `id`s should be unique):
```jsx
<Form action="/signup/insert" method="POST">
    <FormGroup>
        <Input id="username" name="uid" placeholder="Username" /><br />
        <Input id="password" type="password" name="pwd" placeholder="Password" /><br />
        <Input id="email" type="email" name="email" placeholder="E-mail address" /><br />
        <Button id="switch">Sign Up</Button>
     </FormGroup>
 </Form>
```

You may also want to consider using `Label` with `for` matching the input's `id` it is associated with. You can add the `hidden` prop to `Label` to visually hide it. I recommend this for accessibility; screen reader will better be able to identify the input to the user.

Here is an example of your markup using reactstrap (with a few improvements, but visually the same): https://www.webpackbin.com/bins/-Kr1e2_6H_kPxTBe5ar- You can inspect the output and see the rendered `<form>` has the attributes you wanted.

**Comment by sjorsvanheuveln** [comment]
Allright thanks a lot again. And then the button should be like this, I assume:
`<Button type="submit">Log In</Button>`

**Comment by TheSharpieOne** [comment]
The `type` will default to `submit` (at long as there is not an `onClick` prop), so you don't even need that.
You can add `color` to better style it if you want (out-of-the-box bootstrap "brand" values: `primary`, `success`, `info`, `warning`, `danger`, `secondary` (defaults to `secondary`)).

**Comment by hariDasu** [comment]
what about setting headers? when i submit with this form i get "no token provided" of course, because i dont have a place to set the bearer token

**Comment by gergely-nagy** [comment]
@hariDasu you cannot set custom headers in html forms without xhr plugins. You have to use it with ajax.

Popular libs:

- [axios](https://github.com/axios/axios)
- [superagent](https://github.com/visionmedia/superagent)
